<a href="https://colab.research.google.com/github/MOHIT335448/Minor-Project---2/blob/main/Verzeo_Mini_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***ML-MINOR-APRIL***

In [1]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np


In [2]:
df = pd.read_csv('DataCoSupplyChainDataset.csv', encoding= 'unicode_escape')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180519 entries, 0 to 180518
Data columns (total 53 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   Type                           180519 non-null  object 
 1   Days for shipping (real)       180519 non-null  int64  
 2   Days for shipment (scheduled)  180519 non-null  int64  
 3   Benefit per order              180519 non-null  float64
 4   Sales per customer             180519 non-null  float64
 5   Delivery Status                180519 non-null  object 
 6   Late_delivery_risk             180519 non-null  int64  
 7   Category Id                    180519 non-null  int64  
 8   Category Name                  180519 non-null  object 
 9   Customer City                  180519 non-null  object 
 10  Customer Country               180519 non-null  object 
 11  Customer Email                 180519 non-null  object 
 12  Customer Fname                

In [4]:
df.describe()

Days for shipping (real)  ...  Product Status
count             180519.000000  ...        180519.0
mean                   3.497654  ...             0.0
std                    1.623722  ...             0.0
min                    0.000000  ...             0.0
25%                    2.000000  ...             0.0
50%                    3.000000  ...             0.0
75%                    5.000000  ...             0.0
max                    6.000000  ...             0.0

[8 rows x 29 columns]

In [9]:
df

Type  ...   Shipping Mode
0          DEBIT  ...  Standard Class
1       TRANSFER  ...  Standard Class
2           CASH  ...  Standard Class
3          DEBIT  ...  Standard Class
4        PAYMENT  ...  Standard Class
...          ...  ...             ...
180514      CASH  ...  Standard Class
180515     DEBIT  ...    Second Class
180516  TRANSFER  ...  Standard Class
180517   PAYMENT  ...  Standard Class
180518   PAYMENT  ...  Standard Class

[180519 rows x 53 columns]

1) List all the cities and the respective number of pending orders in 'India'?

All the cities are listed below

In [62]:
df[['Customer City' , 'Order City' ]]


Customer City   Order City
0             Caguas       Bekasi
3        Los Angeles   Townsville
6             Caguas    Guangzhou
13            Caguas       Sangli
15            Caguas       Sangli
...              ...          ...
180500    Saint Paul  Rockhampton
180505      Highland      Qingdao
180508       Waipahu    Guangshui
180512        Caguas      Chengdu
180515   Bakersfield     Hirakata

[59491 rows x 2 columns]

In [8]:
df[(df['Order Country'] == 'India') & (df['Order Status'] == 'PENDING')]

Type  ...   Shipping Mode
1       TRANSFER  ...  Standard Class
21      TRANSFER  ...  Standard Class
45      TRANSFER  ...  Standard Class
46      TRANSFER  ...    Second Class
382     TRANSFER  ...  Standard Class
...          ...  ...             ...
180142  TRANSFER  ...  Standard Class
180164  TRANSFER  ...     First Class
180248  TRANSFER  ...    Second Class
180331  TRANSFER  ...  Standard Class
180405  TRANSFER  ...  Standard Class

[537 rows x 53 columns]

the respective number of pending orders in 'India' are 537

2) Which country has the most suspected fraud?

In [14]:
df[(df['Order Status'] == 'SUSPECTED_FRAUD')]

Type  ...   Shipping Mode
10      TRANSFER  ...    Second Class
24      TRANSFER  ...    Second Class
183     TRANSFER  ...  Standard Class
184     TRANSFER  ...  Standard Class
185     TRANSFER  ...  Standard Class
...          ...  ...             ...
180274  TRANSFER  ...    Second Class
180309  TRANSFER  ...     First Class
180352  TRANSFER  ...  Standard Class
180406  TRANSFER  ...  Standard Class
180490  TRANSFER  ...  Standard Class

[4062 rows x 53 columns]

In [15]:
df['Order Country'].mode()

0    Estados Unidos
dtype: object

Estados Unidos is the country which has the most suspected fraud

3. Which product(s) has/have the maximum discount?

In [45]:
df[(df['Order Item Discount']==df['Order Item Discount'].max()) & (df['Order Country'])]

Type  ...   Shipping Mode
12606  DEBIT  ...  Standard Class

[1 rows x 53 columns]

SOLE E35 Elliptical has/have the maximum discount

4. How many successfully processed orders have days for shipping real (Real) > Days for shipment (Scheduled)?

In [56]:
df[df['Days for shipping (real)']>df['Days for shipment (scheduled)']]

Type  ...   Shipping Mode
6       DEBIT  ...     First Class
13      DEBIT  ...     First Class
15      DEBIT  ...     First Class
18      DEBIT  ...     First Class
25      DEBIT  ...    Second Class
...       ...  ...             ...
180479  DEBIT  ...  Standard Class
180499  DEBIT  ...    Second Class
180508  DEBIT  ...    Second Class
180512  DEBIT  ...  Standard Class
180515  DEBIT  ...    Second Class

[34199 rows x 53 columns]

34199 oders have successfully processed orders have days for shipping real (Real) > Days for shipment (Scheduled)
